### ISSUE TITLE
Is Treesitter Overkill for Our Requirements?

### ISSUE THREAD

**Issue by binhtran432k** [cucumber/language-service#182]
### 🤔 What's the problem you've observed?

While Treesitter offers certain parsing capabilities, its current stage of development can introduce challenges that may outweigh its benefits for our specific needs. Notably, the significant size of its WASM parsers (~90% of the language service bundle) seems redundant considering our primary need is simply extracting text strings (easily achievable with Javascript regex).

### ✨ Do you have a proposal for making it better?

I suggest that we use a simple regex solution, such as [VSCucumberAutoComplete](https://github.com/alexkrechik/VSCucumberAutoComplete). Alternatively, we can explore other parser generators like peg.js, langium, lezer (treesitter-like in pure JavaScript), etc.

### 📚 Any additional context?

This solution addresses nearly all issues associated with Treesitter, including those related to non-functionality in Node versions 19 and above.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/developer_experience.md)*

**Comment by helgardferreira** [comment]
Yeah, with the current tree-sitter parsing setup, the language service is falling short of [VSCucumberAutoComplete](https://github.com/alexkrechik/VSCucumberAutoComplete) unfortunately.

I'm not sure if it's the version that this project is using, or what exactly, but the parsing isn't even matching with the [tree-sitter playground](https://tree-sitter.github.io/tree-sitter/playground) currently. For instance, try parsing a typescript file that uses class decorators and you'll see the parser fail arbitrarily.
